src/daft-parquet/src/file.rs

```diff
@@ -82,9 +86,13 @@ where
     type Error = parquet2::error::Error;
     type Item = Page;
     fn advance(&mut self) -> Result<(), Self::Error> {
-        let val = self.handle.block_on(async {
-            let mut s_guard = self.src.lock().await;
-            s_guard.next().await
+        let handle = self.handle.clone();
+        let src = self.src.clone();
+        let val = tokio::task::block_in_place(move || {
```

Comment:
Using `block_in_place` because it is called in a tokio runtime context

Reply:
discussed offline. I think this does what we want it to.